osu.Game/Screens/Edit/EditorLoader.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Screens;
+using osu.Framework.Threading;
+using osu.Game.Beatmaps;
+using osu.Game.Screens.Menu;
+
+namespace osu.Game.Screens.Edit
+{
+    /// <summary>
+    /// Transition screen for the editor.
+    /// Used to avoid backing out to main menu/song select when switching difficulties from within the editor.
+    /// </summary>
+    public class EditorLoader : OsuScreen
+    {
+        public override float BackgroundParallaxAmount => 0.1f;
+
+        public override bool AllowBackButton => false;
+
+        public override bool HideOverlaysOnEnter => true;
+
+        public override bool DisallowExternalBeatmapRulesetChanges => true;
+
+        [Resolved]
+        private BeatmapManager beatmapManager { get; set; }
+
+        [CanBeNull]
+        private ScheduledDelegate scheduledDifficultySwitch;
+
+        protected override void LogoArriving(OsuLogo logo, bool resuming)
+        {
+            base.LogoArriving(logo, resuming);
+            // the push cannot happen in OnEntering() or similar (even if scheduled), because the transition from main menu will look bad.
+            // that is because this screen pushing the editor makes it no longer current, and OsuScreen checks if the screen is current
+            // before enqueueing this screen's LogoArriving onto the logo animation sequence.
+            logo.Delay(300).Schedule(pushEditor);
```

Comment:
Any specific reason for the 300ms delay? I can't seem to see an issue without this present.

Reply:
Uhh, the figure comes from the "basic" animation that the base `LogoArriving()` implementation specifies. But now that I think about it, though, it's not necessary. The only thing that matters is that the push happens in `LogoArriving` after all of the transforms are in place.

Reply:
This delay is now removed with 5b9f377.